### DISCUSSION TITLE
How is your development setup?

### DISCUSSION THREAD

**Discussion by andarotajo** [custom-components/pyscript#558] [Q&A]
I'm an AppDaemon user and currently looking to try out Pyscript to see how compares to AD. I've finished the basic setup including Jupyter but wanted to know what the setups look like for you guys.

- **Do you use Git or any version control to edit your files?**
I have my HA config (including the AppDaemon apps) on a private Git repository. For HA it's very useful as I'm checking the config for validity before pulling it on my instance and for AD I prefer to test the apps I develop before putting them on my HA instance
- **How do you test your code?**
Since AD is a seperate program, I'd run a local instance of it to test out my code before pushing it onto Git. I'd guess that the goto way here is to use Jupyter?
- **How do you organize your scripts/apps?**

Thanks for any feedback :)

**Comment by ALERTua** [comment]
I don't know what Jupyter is and have never used it.
I haven't used AppDaemon.
My main OS is Windows 11, and I use PyCharm.
My /config is shared via Samba and bound to a local logical drive, which I open as a Project in PyCharm.
My Pyscript scripts are located at /config/pyscript.
My Pyscript apps are located at /config/pyscript/apps.
My Pyscript modules are located at /config/pyscript/modules.
My whole /config is being pushed to a private git repository using the Home Assistant Git Exporter addon.
My modules are [shared](https://github.com/ALERTua/ha_pyscript_modules) to GitHub separately.
Code highlighting works correctly thanks to [pyscript_autocomplete](https://github.com/dmamelin/pyscript_autocomplete).

**Reply by andarotajo** [reply]
I'm also using Samba but I go the roundabout way over Git to edit my stuff because I've broken my config once which prevented HA from booting up 😅 

I assume you test your scripts/apps directly on your live system?

**Reply by ALERTua** [reply]
> I assume you test your scripts/apps directly on your live system?

Pyscript scripts cannot prevent HA from starting, as they load after HA starts.
I develop and debug my scripts on my live system.

**Comment by oskretc** [comment]
Using code server (vscode on the web) edidint directly the pyscript files from the config folder

private bitbucket for storing all ha config

**Comment by niklaspeura** [comment]
I have HA running in a docker container on my raspi. I then use vscode and ssh into raspberry and use dev containers extension to access the python interpreter inside the HA container.

What I still don't know if how to run the pyscript stuff from vscode :D

**Comment by FredericG-BE** [comment]
I have a question...

I have files in pyscipt/apps/ and pyscipt/modules/ that I want to have in a git repository for sharing and others in pyscipt/apps/ and pyscipt/modules/ that I would like to keep in a private repository. How could I organize that? 
I cannot have 2 repositories under pyscript I suppose. I could perhaps have the repositories somewhere else and make (soft) links from  pyscipt/apps/ and pyscipt/modules/ to where the repositories are... Are there other approaches?

Thanks
